inspirehep/dojson/hep/rules/bd9xx.py

```diff
@@ -236,20 +236,22 @@ def references2marc(self, key, value):
     journal_artid = get_value(reference, 'publication_info.artid')
     pubnote = build_pubnote(journal_title, journal_volume, journal_pg_start,
                             journal_pg_end, journal_artid)
+    doi = get_value(reference, 'dois[0]')
+
     return {
         '0': get_recid_from_ref(value.get('record')),
         '1': get_value(reference, 'texkey'),
-        'a': get_value(reference, 'dois[0]'),
+        'a': 'doi:{}'.format(doi) if doi else doi,
         'c': get_value(reference, 'collaboration'),
         'e': [a.get('full_name') for a in reference.get('authors', [])
               if a.get('role') == 'ed.'],
         'h': [a.get('full_name') for a in reference.get('authors', [])
               if a.get('role') != 'ed.'],
         'm': get_value(reference, 'misc'),
-        'o': get_value(reference, 'number'),
+        'o': get_value(reference, 'label'),
         'i': get_value(reference, 'publication_info.isbn'),
         'p': get_value(reference, 'imprint.publisher'),
-        'r': ['arXiv:' + el if el and is_arxiv(el) else el for el in repnos],
+        'r': get_value(reference, 'arxiv_eprint'),
```

Comment:
This requires the addition of the `arXiv:` prefix, see https://michamos.github.io/inspire-schemas/records/elements/reference.html#arxiv-eprint.